### ISSUE TITLE
Idea: Exported function to hide column name

### ISSUE THREAD

**Issue by Zangomangu** [OvileAmriam/MTA-Beautify-Library#24]
Would be nice if you could hide the column name, eg. if you put a gridlist inside a deckpane, the deck name works as a column name.

Like this, where "Column" can be toggled

![image](https://user-images.githubusercontent.com/59518113/132579075-91b2af4c-aff7-4771-9ad4-e9bbed7e232a.png)

**Comment by ov-tron** [comment]
@Zangomangu What about multiple columns then?

**Comment by ov-tron** [comment]
Maybe add exported function to hide column completely for grid list? otherwise this could be issue for multiple columns probably, Actually this  was the job of combo box which is pending yet tbh; assuming you are referring to that

**Comment by ov-aviril** [comment]
![image](https://user-images.githubusercontent.com/89867985/132590616-e45a0add-9c29-4447-b91f-29e877b24055.png)
I think Zango probably wants this in the form of comobox, which is basically deck + gridlist but with single column

**Comment by Zangomangu** [comment]
> @Zangomangu What about multiple columns then?

Yeah it should hide all the columns, basically the top row that has the column names.



> Maybe add exported function to hide column completely for grid list? otherwise this could be issue for multiple columns probably, Actually this was the job of combo box which is pending yet tbh; assuming you are referring to that

Yeah. But to me a combobox is different, if you can hide the column names, a gridlist inside a deckpane is really neat for some stuff.

Is there any way currently to create a section row in a gridlist? It's basically just a row that has a different font to sort of separate the gridlist like a category

**Comment by ov-tron** [comment]
> > @Zangomangu What about multiple columns then?
> 
> Yeah it should hide all the columns, basically the top row that has the column names.
> 
> > Maybe add exported function to hide column completely for grid list? otherwise this could be issue for multiple columns probably, Actually this was the job of combo box which is pending yet tbh; assuming you are referring to that
> 
> Yeah. But to me a combobox is different, if you can hide the column names, a gridlist inside a deckpane is really neat for some stuff.
> 
> Is there any way currently to create a section row in a gridlist? It's basically just a row that has a different font to sort of separate the gridlist like a category

Can you show some visual reference about gridlist sections?

**Comment by Zangomangu** [comment]
![image](https://user-images.githubusercontent.com/59518113/132940125-7f3ce52b-17c2-4aef-9799-958f82dfa584.png)

"Truckers" and "Fighters" are sections

**Comment by ov-tron** [comment]
Ah thats easy to add, can you give a syntax for adding section? I think when adding rows it should be passing section id too that way or how does it works to know which row is under which?

**Comment by Zangomangu** [comment]
Could add a parameter to addRow that determines if it's a section.

`local row = beautify.gridlist.addRow(gridlist, true)`

A section row shouldnt be selectable

**Comment by ov-tron** [comment]
Yeah tho i meant like if you want to add a non section row in 2nd or 3rd section how would you determine that?

**Comment by Zangomangu** [comment]
Well you would have to add the rows in the right order. I see what you mean though. 
If you add ordering to columns, then it will mess up the sections, unless rows belong under a section.

If rows belong under a section then you could have
`local rowSectionID = beautify.gridlist.addRowSection(gridlist)`
and
`local rowID = beautify.gridlist.addRow(gridlist, rowSectionID)`

Then if column ordering is implemented, it can first sort the sections alphabetically, then the rows under the sections alphabetically.
